### PR TITLE
Select: fix slice of value when value is undefined or null

### PR DESCRIPTION
**Pull request by luckyCao**
check if value is empty before slice it 
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by luckyCao** [comment]
Thanks remind

**Comment by luckyCao** [comment]
@wacky6 unit test added

File: test/unit/specs/select.spec.js [hunk 1]
```diff
@@ -821,6 +821,47 @@ describe('Select', () => {
...
+            value: '选项2',
+            label: '双皮奶'
+          }],
+          value: undefined
+        };
+      }
+    }, true);
+
+    vm.value = null;
+    vm.$nextTick(_ => {
```
Comment (ziyoung) [inline test/unit/specs/select.spec.js]: 使用 async 函数。

**Comment by ziyoung** [comment]
@luckyCao 代码冲突了。

**Pull request merged**